java/src/test/java/ai/rapids/cudf/ReductionTest.java

```diff
@@ -77,186 +81,356 @@ private static Scalar buildExpectedScalar(ReductionAggregation op, DType baseTyp
+        Lists.newArrayList(60f, 100f, Float.NaN, null)
+    };
+    HostColumnVector.DataType fpList = new HostColumnVector.ListType(
+        true, new HostColumnVector.BasicType(true, DType.FLOAT32));
+    return Stream.of(
+        Arguments.of(ReductionAggregation.mergeLists(), inputs, fpList,
+            new Float[]{-1f, 7f, null,
+                7f, 50f, 60f, Float.NaN,
+                60f, 100f, Float.NaN, null}, 0f),
+        Arguments.of(ReductionAggregation.mergeSets(NullEquality.EQUAL, NaNEquality.ALL_EQUAL),
+            inputs, fpList,
+            new Float[]{-1f, 7f, 50f, 60f, 100f, Float.NaN, null}, 0f),
+        Arguments.of(ReductionAggregation.mergeSets(NullEquality.UNEQUAL, NaNEquality.ALL_EQUAL),
+            inputs, fpList,
+            new Float[]{-1f, 7f, 50f, 60f, 100f, Float.NaN, null, null}, 0f),
+        Arguments.of(ReductionAggregation.mergeSets(NullEquality.EQUAL, NaNEquality.UNEQUAL),
+            inputs, fpList,
+            new Float[]{-1f, 7f, 50f, 60f, 100f, Float.NaN, Float.NaN, null}, 0f),
+        Arguments.of(ReductionAggregation.mergeSets(),
+            inputs, fpList,
+            new Float[]{-1f, 7f, 50f, 60f, 100f, Float.NaN, Float.NaN, null, null}, 0f)
     );
   }
 
   private static void assertEqualsDelta(ReductionAggregation op, Scalar expected, Scalar result,
-                                        Double percentage) {
+      Double percentage) {
     if (FLOAT_REDUCTIONS.contains(op.getWrapped().kind)) {
       assertEqualsWithinPercentage(expected.getDouble(), result.getDouble(), percentage);
+    } else if (expected.getType().typeId == DType.DTypeEnum.LIST) {
+      try (ColumnView e = expected.getListAsColumnView();
+           ColumnView r = result.getListAsColumnView()) {
+        AssertUtils.assertColumnsAreEqual(e, r);
+      }
     } else {
       assertEquals(expected, result);
     }
   }
 
   private static void assertEqualsDelta(ReductionAggregation op, Scalar expected, Scalar result,
-                                        Float percentage) {
+      Float percentage) {
     if (FLOAT_REDUCTIONS.contains(op.getWrapped().kind)) {
       assertEqualsWithinPercentage(expected.getFloat(), result.getFloat(), percentage);
+    } else if (expected.getType().typeId == DType.DTypeEnum.LIST) {
+      try (ColumnView e = expected.getListAsColumnView();
+           ColumnView r = result.getListAsColumnView()) {
+        AssertUtils.assertColumnsAreEqual(e, r);
+      }
     } else {
       assertEquals(expected, result);
     }
   }
 
   @ParameterizedTest
   @MethodSource("createBooleanParams")
-  void testBoolean(ReductionAggregation op, Boolean[] values, Object expectedObject, Double delta) {
-    try (Scalar expected = buildExpectedScalar(op, DType.BOOL8, expectedObject);
+  void testBoolean(ReductionAggregation op, Boolean[] values,
+      HostColumnVector.DataType tpe, Object expectedObject, Double delta) {
+    try (Scalar expected = buildExpectedScalar(op, tpe, expectedObject);
```

Comment:
Are all of the occurrences of `tpe` supposed to be `type`?

Reply:
Done.